### ISSUE TITLE
There is no any ability to customize size of inputs in DatePicker.RangePicker

### ISSUE THREAD

**Issue by lini-lini** [ant-design/ant-design#47397]
### What problem does this feature solve?

Right now the one possible way to customize size of inputs (to display them on the start of the line, for example) is to cut it by setting width for .ant-picker-input class.
And due to the reason that inputs inside of the DatePicker.RangePicker have 'size' attribute which is set to 12, we can't use dynamic width characteristics ('fit-content', for example).
The problem is also actual for cases, when a developer wants to set a custom date pattern which can have more symbols, than the actual size.
![image](https://github.com/ant-design/ant-design/assets/152303565/c40287a5-5a13-44f5-af11-ef47c046c39b)

In conclusion, if inputs have this 'size' attribute, it would be nice to have an ability to change attribute 'size' of inputs inside of DatePicker.RangePicker. It would solve the issue with static width and give an ability for more various customizations.

### What does the proposed API look like?

Add 'inputSize' argument to DatePicker.RangePicker which will update the HTML attribute 'size' of inputs inside Range

**Comment by Roxannej** [comment]
you can config you width like <RangePicker showTime style={{ width: 350 }} />

**Comment by EmilyyyLiu** [comment]
You can now customize each input's width via the semantic `styles`, where the `input` slot applies directly to the `<input>`:

```tsx
<DatePicker.RangePicker
  showTime
  styles={{ input: { width: 'fit-content' } }}
/>
```

If you just need a wider picker overall, using the root style also works: `<RangePicker showTime style={{ width: 350 }} />`. With `styles.input` you don't need a global `.ant-picker-input` override and can override the static width coming from the HTML `size` attribute. See https://ant.design/components/date-picker#semantic-dom for the full list.

Can we close this issue? If there are still cases not covered, feel free to continue the discussion.
